shifthappens/tasks/imagenet_metashift/imagenet_metashift.py

```diff
@@ -0,0 +1,360 @@
+)
+
+
+@sh_benchmark.register_task(
+    name="ImageNet-MetaShift",
+    relative_data_folder="imagenet_metashift",
+    standalone=True,
+)
+@dataclasses.dataclass
+class ImageNetMetaShift(Task):
+    """Evaluate the classification accuracy on a small subset of the ImageNet-MetaShift dataset.
+    Each class in the test case contains several subsets, representing distribution of images in different contexts.
+
+    [1] We have matched the labels in ImageNet-1k to MetaShift. Since ImageNet-1k has heterogeneous hierarchy,
+        a class can have many breeds. Take dog as an example, MetaShift only contains one class of dog,
+        while ImageNet has many kinds of dogs. In our metrics, any results under dog hierarchy are viewed as correct
+        when evaluate the classification results of dog.
+    """
+
+    resources = [
+        (
+            "images.zip",
+            "https://nlp.stanford.edu/data/gqa/images.zip",
+        )
+    ]
+
+    def parse_node_str(self, node_str):
+        """parse the node into class and context string"""
+        tag = node_str.split("(")[-1][:-1]
+        subject_str = node_str.split("(")[0].strip()
+        return subject_str, tag
+
+    def load_candidate_subsets(self):
+        """
+        load candidate subsets for generating
+        """
+        pkl_save_path = os.path.join(
+            self.data_root, "imagenet-metashift", "selected-candidate-subsets.pkl"
+        )
+        with open(pkl_save_path, "rb") as pkl_f:
+            load_data = pickle.load(pkl_f)
+            print("pickle load", len(load_data), pkl_save_path)
+            return load_data
+
+    def copy_image_for_subject(
+        self, root_folder, subject_str, subject_data, node_name_to_img_id, trainsg_dupes
+    ):
+        """Copy Image Sets: Work at subject_str level"""
+
+        # Iterate all the subsets of the given subject
+        for node_name in subject_data:
+            subject_str, tag = self.parse_node_str(node_name)
+
+            # Create dataset a new folder
+            subject_localgroup_folder = os.path.join(
+                root_folder, subject_str, node_name
+            )
+            if os.path.isdir(subject_localgroup_folder):
+                shutil.rmtree(subject_localgroup_folder)
+            os.makedirs(subject_localgroup_folder, exist_ok=False)
+
+            for image_idx_in_set, imageID in enumerate(
+                node_name_to_img_id[node_name] - trainsg_dupes
+            ):
+
+                src_image_path = os.path.join(
+                    self.data_root,
+                    "imagenet-metashift",
+                    "allImages",
+                    "images",
+                    imageID + ".jpg",
+                )
+                dst_image_path = os.path.join(
+                    subject_localgroup_folder, imageID + ".jpg"
+                )
+                shutil.copyfile(src_image_path, dst_image_path)
```

Comment:
```suggestion
                os.symlink(os.path.abspath(src_image_path), os.path.abspath(dst_image_path))
```
I would suggest replacing explicit copying images with creating symbolic links.
This requires less space and reduces the time for task evaluation significantly.